lib/pbench/test/unit/server/query_apis/test_index_mappings.py

```diff
@@ -108,9 +108,14 @@ def test_with_no_index_document(self, client, server_config):
         present in the database.
         """
         with client:
-            response = client.get(f"{server_config.rest_uri}/index/mappings/test")
-            assert response.status_code == HTTPStatus.NOT_FOUND
-            assert response.json["message"] == "Mapping not found"
+            response = client.get(
+                f"{server_config.rest_uri}/index/mappings/bad_data_object_type"
+            )
+            assert response.status_code == HTTPStatus.BAD_REQUEST
+            assert (
+                response.json["message"]
+                == "Unrecognized keyword ['bad_data_object_type'] given for parameter index_key; allowed keywords are ['iterations', 'search', 'timeseries']"
```

Comment:
This is a nit, but do we want to name the parameter "index_key"?  To the extent that this name is externally visible and subject to upward compatibility constraints, shouldn't we name it something that isn't tied to the particular database implementation that we happen to be using at the moment?

Reply:
Yeah that's a valid point, but this is raised because the `KeywordError` is calculating all the valid keywords for the key in json where it failed. And we validate the URI key by making it a json schema, to change the behavior I guess we will have to update the way we do the validation of the URI parameter.

Reply:
My concern is not about what this message represents on the Server side of the interface but rather what the client (particularly the human) will see on the its side of the interface.

Most of the text of that message is fine ("Unrecognized keyword" is good; listing the allowable keywords is excellent; displaying the offending keyword is good too); what I'm objecting to is just "index_key" -- I think we should find something other than "index" to use in that token, because "index" is too Elasticsearch-y.

Something like "object_type_key" (although, that's a little long) would be good.  ("Data type"?  "Table type"? Something in that vein.)

Reply:
We're talking about a "view", or a "slice", of the dataset, not really even a "data type". I agree I'd prefer to avoid referring to this as an "index" since our Elasticsearch document schema should be a backend implementation detail.

I think that in an ideal REST model, where "dataset" is a resource, we might have something more like

* `GET /datasets/<id>?view=overview` for the current `/datasets/detail`, (or perhaps `/datasets/<id>/overview`?),
* `GET /datasets/<id>?view=iterations`,
* `GET /datasets/<id>?view=samples`

If I hadn't faced the issue of trying to fit this into the dashboard, I probably would have just gone that way instead of the muddle we've got now, which grew out of the current dashboard service/model structure. Right now, I'm content to get a working set of comprehensive APIs that we can use and analyze and then throw out in favor of something better...

But for right now, I think I like "view" better than any other alternative here.

Reply:
OK, given that the currently proposed endpoint is `/index/mappings/<string:index_key>`, I guess that having the parameter name include "index" is OK, since the base component of the URI is also "index".

I thought we had decided to change the base to "dataset", but apparently I'm confused.  So, if we're not changing the resource name then we don't need to change the parameter name.  🤷‍♂️